### PR TITLE
Run `dapp-test` through nix-shell in ci

### PR DESCRIPTION
**Pull request by MrChico**
Ensures `dapp-tests` are run through travis, removes the ds-test submoule which should set the stage for #377

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ script:
     [[ "$TRAVIS_OS_NAME" == "osx" ]] && PLATFORM=darwin || PLATFORM=linux
 
     nix-build -j auto release.nix -A dapphub.$PLATFORM.stable &&
+    cd src/dapp-test && nix-shell --command 'make' --pure && cd ../..
```
Comment (asymmetric) [inline .travis.yml:23]: You can just pass the `shell.nix` path to `nix-shell`, without `cd`ing.

File: src/seth/default.nix [hunk 2]
```diff
@@ -15,8 +15,7 @@ stdenv.mkDerivation rec {
   postInstall =
     let
       path = lib.makeBinPath [
-        bc coreutils curl ethsign git gnused jshon nodejs perl
-        bc coreutils curl ethsign git gnused hevm jshon nodejs perl
+        bc coreutils curl ethsign git gnused hevm jshon nodejs perl jq
```
Comment (asymmetric) [inline src/seth/default.nix:18]: Can we sort this bad boy plz?

File: src/dapp-tests/shell.nix [hunk 3]
```diff
@@ -0,0 +1,8 @@
+let dapptools = import ../../. {};
+in dapptools.mkShell {
+  buildInputs = with dapptools; [
+  dapp
```
Comment (asymmetric) [inline src/dapp-tests/shell.nix:4]: Plz indent line

File: src/dapp-tests/shell.nix [hunk 4]
```diff
@@ -0,0 +1,8 @@
+let dapptools = import ../../. {};
```
Comment (asymmetric) [inline src/dapp-tests/shell.nix:1]: I would just call this `pkgs`, since it's actually all of nixpkgs.

File: src/dapp-tests/Makefile [hunk 5]
```diff
@@ -1,3 +1,7 @@
-all:; ./run-tests
-
+all:;
+	dapp install ds-test
+	./run-tests
+	dapp uninstall ds-test
+	git reset HEAD^^
```
Comment (asymmetric) [inline src/dapp-tests/Makefile:5]: This is not super pretty. It also reinforces the coupling between dapp and git.

I'll see if making a derivation out of `ds-test` works.

Comment (MrChico) [reply src/dapp-tests/Makefile:5]: I wonder if this can be made with the nix function `solidityPaclage`

**Pull request closed without merging**